src/main/java/co/inventorsoft/StreamHomework.java

```diff
@@ -49,19 +55,34 @@ public Map<String, User> groupByEmail(final List<User> users) {
      * @return map {age : people with this age}
      */
     public Map<Integer, List<Person>> groupByAge(final List<Person> people) {
-        return null;
+        Map<Integer, List<Person>> userMap = new HashMap<>();
+        people.forEach(person -> {
+            List<Person> peopleThisAge = people.stream()
+                    .filter(person1 -> Objects.equals(person1.getAge(), person.getAge()))
+                    .collect(Collectors.toList());
+            userMap.put(person.getAge(), peopleThisAge);
+        });
+        return userMap;
     }
 
     /**
      * Creates single string, representing all people names, emphasizing uniqueness!
      * Example:
-     *    input:  [{13, "Harry"}, {13, "Ron"}, {14, "Hermione"}, {13, "Harry"}]
-     *    output: "Distinct names: Harry, Ron, Hermione!"
+     * input:  [{13, "Harry"}, {13, "Ron"}, {14, "Hermione"}, {13, "Harry"}]
+     * output: "Distinct names: Harry, Ron, Hermione!"
      *
      * @param people collection of people
      * @return string with unique names, like "Distinct names: a, b, c!"
      */
     public String collectDistinctNames(final List<Person> people) {
-        return null;
+        List<String> allNames = new ArrayList<>();
+        people.forEach(person -> allNames.add(person.getName()));
+        StringBuilder names = new StringBuilder("Distinct names: ");
+        allNames.stream()
+                .filter(name -> names.indexOf(name) < 0)
+                .forEach(name -> names.append(name).append(", "));
+        names.replace(names.length() - 2, names.length(), "!");
```

Comment:
Again, there's much simpler way to get the same result.
Look into other collectors. You might also find a hint in the presentation.